src/Aspire.Hosting.MySql/MySqlBuilderExtensions.cs

```diff
@@ -28,7 +29,7 @@ public static IResourceBuilder<MySqlServerResource> AddMySql(this IDistributedAp
         var passwordParameterName = $"{name}-password";
         var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, passwordParameterName);
 
-        if (passwordParameter.Default is not null)
+        if (builder.Environment.IsDevelopment() && builder.ExecutionContext.IsRunMode && passwordParameter.Default is not null)
```

Comment:
Why does the environment matter?

Reply:
User secrets is only added as a configuration provider when the environment is Development

Reply:
I removed the environment check @mitchdenny